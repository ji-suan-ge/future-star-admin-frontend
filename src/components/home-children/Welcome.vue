<template>
  <el-carousel :interval="4000" type="card" height="500px">
    <el-carousel-item v-for="imageAddress in imageAddressArray" :key="imageAddress.id">
      <el-image :src="imageAddress.address" fit="fill">
        <div slot="error" class="image-slot">
          <i class="el-icon-picture-outline"></i>
        </div>
      </el-image>
    </el-carousel-item>
  </el-carousel>
</template>

<script>
export default {
  name: 'Welcome',
  data () {
    return {
      imageAddressArray: [
        {
          id: '1',
          address: require('../../assets/backgroundImage1.jpg')
        },
        {
          id: '2',
          address: require('../../assets/backgroundImage2.jpg')
        },
        {
          id: '3',
          address: require('../../assets/backgroundImage3.jpg')
        },
        {
          id: '4',
          address: require('../../assets/backgroundImage4.jpg')
        }
      ]
    }
  }
}
</script>

<style scoped>

</style>
