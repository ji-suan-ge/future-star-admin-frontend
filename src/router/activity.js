import CreateActivityPage from '../page/CreateActivityPage'
import ShowActivityParticipantsPage from '../page/ShowActivityParticipantsPage'
import EditExistingActivityPage from '../page/EditExistingActivityPage'
import ShowActivityInformationPage from '../page/ShowActivityInformationPage'

export default [
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  },
  {
    path: '/show-participants',
    name: 'ShowActivityParticipantsPage',
    component: ShowActivityParticipantsPage
  },
  {
    path: '/edit-activity',
    name: 'EditExistingActivityPage',
    component: EditExistingActivityPage
  },
  {
    path: '/show-activity-details',
    name: 'ShowActivityInformationPage',
    component: ShowActivityInformationPage
  }
]
