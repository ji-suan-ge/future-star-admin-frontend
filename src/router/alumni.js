import ShowAlumniInformationPage from '../page/alumni-management/ShowAlumniInformationPage'
import EditAlumniInformationPage from '../page/alumni-management/EditAlumniInformationPage'
import ShowAlumniEditHistoryPage from '../page/alumni-management/ShowAlumniEditHistoryPage'
import ShowAlumniHistoryInformationPage from '../page/alumni-management/ShowAlumniHistoryInformationPage'

export default [
  {
    path: '/show-all-information',
    name: 'ShowAlumniInformationPage',
    component: ShowAlumniInformationPage
  },
  {
    path: '/edit-alumni-information',
    name: 'EditAlumniInformationPage',
    component: EditAlumniInformationPage
  },
  {
    path: '/show-edit-history',
    name: 'ShowAlumniEditHistoryPage',
    component: ShowAlumniEditHistoryPage
  },
  {
    path: '/show-history-information',
    name: 'ShowAlumniHistoryInformationPage',
    component: ShowAlumniHistoryInformationPage
  }
]
