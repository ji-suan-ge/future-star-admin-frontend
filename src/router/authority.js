import AddAdministratorPage from '../page/authority-management/AddAdministratorPage'

export default [
  {
    path: '/add-administrator',
    name: 'AddAdministratorPage',
    component: AddAdministratorPage
  }
]
