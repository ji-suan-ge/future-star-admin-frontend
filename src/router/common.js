import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import homeChildren from './home-children'

export default [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      ...homeChildren
    ]
  }
]
