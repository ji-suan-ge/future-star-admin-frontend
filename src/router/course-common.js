import DetailOfSemesterPage from '../page/course-common/DetailOfSemesterPage'
import DetailOfClassPage from '../page/course-common/DetailOfClassPage'
import DetailOfCoursePage from '../page/course-common/DetailOfCoursePage'
import DetailOfTeacherPage from '../page/course-common/DetailOfTeacherPage'
import DetailOfSubCoursePage from '../page/course-common/DetailOfSubCoursePage'
import EditSemesterInfoPage from '../page/course-common/EditSemesterInfoPage'
import EditClassInfoPage from '../page/course-common/EditClassInfoPage'
import EditCourseInfoPage from '../page/course-common/EditCourseInfoPage'
import EditTeacherInfoPage from '../page/course-common/EditTeacherInfoPage'
import AddCoursePage from '../page/course-common/AddCoursePage'
import AddCourseResourcePage from '../page/course-common/AddCourseResourcePage'
import StudentInClass from '../page/course-common/StudentInClass'

export default [
  {
    path: '/semester-detail',
    name: 'DetailOfSemesterPage',
    component: DetailOfSemesterPage
  },
  {
    path: '/class-detail',
    name: 'DetailOfClassPage',
    component: DetailOfClassPage
  },
  {
    path: '/course-detail',
    name: 'DetailOfCoursePage',
    component: DetailOfCoursePage
  },
  {
    path: '/teacher-detail',
    name: 'DetailOfTeacherPage',
    component: DetailOfTeacherPage
  },
  {
    path: '/sub-course-detail',
    name: 'DetailOfSubCoursePage',
    component: DetailOfSubCoursePage
  },
  {
    path: '/edit-semester-info',
    name: 'EditSemesterInfoPage',
    component: EditSemesterInfoPage
  },
  {
    path: '/edit-class-info',
    name: 'EditClassInfoPage',
    component: EditClassInfoPage
  },
  {
    path: '/edit-course-info',
    name: 'EditCourseInfoPage',
    component: EditCourseInfoPage
  },
  {
    path: '/edit-teacher-info',
    name: 'EditTeacherInfoPage',
    component: EditTeacherInfoPage
  },
  {
    path: '/add-course',
    name: 'AddCoursePage',
    component: AddCoursePage
  },
  {
    path: '/add-course-resource',
    name: 'AddCourseResourcePage',
    component: AddCourseResourcePage
  },
  {
    path: '/student-in-class',
    name: 'StudentInClass',
    component: StudentInClass
  }
]
