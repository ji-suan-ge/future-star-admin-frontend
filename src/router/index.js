import Vue from 'vue'
import VueRouter from 'vue-router'
import LoginPage from '../page/LoginPage'
import HomePage from '../page/HomePage'
import EnrollmentManagementPage from '../page/EnrollmentManagementPage'
import CourseManagementPage from '../page/CourseManagementPage'
import ActivityManagementPage from '../page/ActivityManagementPage'
import AlumniManagementPage from '../page/AlumniManagementPage'
import AuthorityManagementPage from '../page/AuthorityManagementPage'
import CreateActivityPage from '../page/CreateActivityPage'
import AddAdministrator from '../page/AddAdministrator'
import WelcomePage from '../page/WelcomPage'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    name: 'Root',
    redirect: '/login'
  },
  {
    path: '/login',
    name: 'LoginPage',
    component: LoginPage
  },
  {
    path: '/home',
    name: 'HomePage',
    component: HomePage,
    redirect: '/home/welcome',
    children: [
      {
        path: 'welcome',
        name: 'WelcomePage',
        component: WelcomePage
      },
      {
        path: 'enrollment-management',
        name: 'EnrollmentManagementPage',
        component: EnrollmentManagementPage
      },
      {
        path: 'course-management',
        name: 'CourseManagementPage',
        component: CourseManagementPage
      },
      {
        path: 'activity-management',
        name: 'ActivityManagementPage',
        component: ActivityManagementPage
      },
      {
        path: 'alumni-management',
        name: 'AlumniManagementPage',
        component: AlumniManagementPage
      },
      {
        path: 'authority-management',
        name: 'AuthorityManagementPage',
        component: AuthorityManagementPage
      },
      {
        path: 'add-administrator',
        name: 'AddAdministrator',
        component: AddAdministrator
      }
    ]
  },
  {
    path: '/create-activity',
    name: 'CreateActivityPage',
    component: CreateActivityPage
  }
]

const router = new VueRouter({
  routes
})

export default router
