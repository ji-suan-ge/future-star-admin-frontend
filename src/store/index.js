import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    activeIndexOfNavigation: '0',
    activeNameOfCourse: 'first',
    activeNameOfEnrollment: 'first',
    activeNameOfActivity: 'first',
    activeSemesterOfCourse: '',
    semesters: '',
    currentSemester: '',
    classes: '',
    currentClass: '',
    courses: '',
    currentCourse: '',
    infoOfAnAlumni: null
  },
  mutations: {
    changeActiveIndexOfNavigation (state, activeIndex) {
      state.activeIndexOfNavigation = activeIndex
    },
    changeActiveNameOfEnrollment (state, activeName) {
      state.activeNameOfEnrollment = activeName
    },
    changeActiveNameOfCourse (state, activeName) {
      state.activeNameOfCourse = activeName
    },
    changeActiveNameOfActivity (state, activeName) {
      state.activeNameOfActivity = activeName
    },
    changeActiveSemesterOfCourse (state, activeID) {
      state.activeSemesterOfCourse = activeID
    },
    changeSemesters (state, semesters) {
      state.semesters = semesters
    },
    changeInfoOfAnAlumni (state, alumniInfo) {
      state.infoOfAnAlumni = alumniInfo
    },
    changeCurrentSemester (state, semester) {
      state.currentSemester = semester
    },
    changeClasses (state, classes) {
      state.classes = classes
    },
    changeCurrentClass (state, currentClass) {
      state.currentClass = currentClass
    },
    changeCourses (state, courses) {
      state.courses = courses
    },
    changeCurrentCourse (state, currentCourse) {
      state.currentCourse = currentCourse
    },
    initializeStateWhileLogin (state) {
      state.activeIndexOfNavigation = '0'
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    },
    initializeStateWhileSwitch (state) {
      state.activeNameOfCourse = 'first'
      state.activeNameOfEnrollment = 'first'
      state.activeNameOfActivity = 'first'
      state.activeSemesterOfCourse = ''
      state.semesters = ''
      state.currentSemester = ''
      state.classes = ''
      state.currentClass = ''
      state.courses = ''
      state.currentCourse = ''
      state.infoOfAnAlumni = null
    }
  },
  actions: {
    changeActiveIndexOfNavigation (context, activeIndex) {
      context.commit('changeActiveIndexOfNavigation', activeIndex)
    },
    changeActiveNameOfEnrollment (context, activeName) {
      context.commit('changeActiveNameOfEnrollment', activeName)
    },
    changeActiveNameOfCourse (context, activeName) {
      context.commit('changeActiveNameOfCourse', activeName)
    },
    changeActiveNameOfActivity (context, activeName) {
      context.commit('changeActiveNameOfActivity', activeName)
    },
    changeActiveSemesterOfCourse (context, activeID) {
      context.commit('changeActiveSemesterOfCourse', activeID)
    },
    changeSemesters (context, semesters) {
      context.commit('changeSemesters', semesters)
    },
    changeInfoOfAnAlumni (context, alumniInfo) {
      context.commit('changeInfoOfAnAlumni', alumniInfo)
    },
    changeCurrentSemester (context, semester) {
      context.commit('changeCurrentSemester', semester)
    },
    changeClasses (context, classes) {
      context.commit('changeClasses', classes)
    },
    changeCurrentClass (context, currentClass) {
      context.commit('changeCurrentClass', currentClass)
    },
    changeCourses (context, courses) {
      context.commit('changeCourses', courses)
    },
    changeCurrentCourse (context, currentCourse) {
      context.commit('changeCurrentCourse', currentCourse)
    },
    initializeStateWhileLogin (context) {
      context.commit('initializeStateWhileLogin')
    },
    initializeStateWhileSwitch (context) {
      context.commit('initializeStateWhileSwitch')
    }
  },
  getters: {
    getActiveIndexOfNavigation (state) {
      return state.activeIndexOfNavigation
    },
    getActiveNameOfEnrollment (state) {
      return state.activeNameOfEnrollment
    },
    getActiveNameOfCourse (state) {
      return state.activeNameOfCourse
    },
    getActiveNameOfActivity (state) {
      return state.activeNameOfActivity
    },
    getActiveSemesterOfCourse (state) {
      return state.activeSemesterOfCourse
    },
    getSemesters (state) {
      return state.semesters
    },
    getInfoOfAnAlumni (state) {
      return state.infoOfAnAlumni
    },
    getCurrentSemester (state) {
      return state.currentSemester
    },
    getClasses (state) {
      return state.classes
    },
    getCurrentClass (state) {
      return state.currentClass
    },
    getCourses (state) {
      return state.courses
    },
    getCurrentCourse (state) {
      return state.currentCourse
    }
  },
  modules: {
  }
})
