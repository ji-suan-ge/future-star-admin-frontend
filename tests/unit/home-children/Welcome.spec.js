import { createLocalVue, mount } from '@vue/test-utils'
import Welcome from '../../../src/components/home-children/Welcome'
import Vuex from 'vuex'
import ElementUI from 'element-ui'
import VueRouter from 'vue-router'
import store from '../../../src/store'
import router from '../../../src/router'

describe('Welcome', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(ElementUI)
  localVue.use(VueRouter)
  const wrapper = mount(Welcome, {
    store,
    router,
    localVue
  })
  test('快照测试', async () => {
    await expect(wrapper.html()).toMatchSnapshot()
  })
  test('是一个 Vue 实例', async () => {
    await expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
