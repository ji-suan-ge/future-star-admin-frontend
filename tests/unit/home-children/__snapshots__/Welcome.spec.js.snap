// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome 快照测试 1`] = `
<div class="el-carousel el-carousel--horizontal">
  <div class="el-carousel__container" style="height: 500px;"><button type="button" class="el-carousel__arrow el-carousel__arrow--left" style="display: none;"><i class="el-icon-arrow-left"></i></button><button type="button" class="el-carousel__arrow el-carousel__arrow--right" style="display: none;"><i class="el-icon-arrow-right"></i></button>
    <div class="el-carousel__item is-active is-animating" style="transform: translateX(0px) scale(1); webkit-transform: translateX(0px) scale(1);">
      <!---->
      <div class="el-image">
        <div class="el-image__placeholder"></div>
        <!---->
      </div>
    </div>
    <div class="el-carousel__item" style="transform: translateX(0px) scale(1); webkit-transform: translateX(0px) scale(1);">
      <!---->
      <div class="el-image">
        <div class="el-image__placeholder"></div>
        <!---->
      </div>
    </div>
    <div class="el-carousel__item" style="transform: translateX(0px) scale(1); webkit-transform: translateX(0px) scale(1);">
      <!---->
      <div class="el-image">
        <div class="el-image__placeholder"></div>
        <!---->
      </div>
    </div>
  </div>
  <ul class="el-carousel__indicators el-carousel__indicators--horizontal">
    <li class="el-carousel__indicator el-carousel__indicator--horizontal is-active"><button class="el-carousel__button">
        <!----></button></li>
    <li class="el-carousel__indicator el-carousel__indicator--horizontal"><button class="el-carousel__button">
        <!----></button></li>
    <li class="el-carousel__indicator el-carousel__indicator--horizontal"><button class="el-carousel__button">
        <!----></button></li>
  </ul>
</div>
`;
